File: Algorithm/CodingTest/src/Programmers/level1/GCD.java
package Programmers.level1;

// https://programmers.co.kr/learn/courses/30/lessons/12940

public class GCD {
    public int gcd(int a, int b) {
        while(b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }

    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        answer[0] = gcd(n,m);
        answer[1] = (n * m) / answer[0];
        return answer;
    }
}

File: Algorithm/Algorithm_py/programmers1/GCD.py
# https://programmers.co.kr/learn/courses/30/lessons/12940?language=python3
def solution(n,m):
  answer = []
  
  for i in range(min(n,m), 0, -1):
    if n % i == 0 and m % i == 0:
      answer.append(i)
      break
  
  for i in range(max(n,m), (n*m) + 1):
    if i % n == 0 and i % m == 0:
      answer.append(i)
      break
  
  return answer